### ISSUE TITLE
LoopStep.java is not entreprise enough

### ISSUE THREAD

**Issue by juliengrenier** [EnterpriseQualityCoding/FizzBuzzEnterpriseEdition#238]
LoopStep doesn't allow to change the increment of the loop. Clearly that's an oversight and it should be fix. LoopStep.java should be renamed OneStepLoopStep and override an abstract class called AbstractLoopStep.

**Comment by emiln** [comment]
It's actually even worse—[LoopStep](/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition/blob/6f865cc/src/main/java/com/seriouscompany/business/java/fizzbuzz/packagenamingpackage/impl/loop/LoopStep.java) is not an implementation of a generalized concept expressed as an interface. It's just a class. This absolutely warrants further investigation.

**Comment by juliengrenier** [comment]
Well it is a given that AbstractLoopStep should implement a LoopStepInterface (or is it a ILoopStep). Also should we open an another ticket to ensure that all classes are implementing an interface?

**Comment by galves** [comment]
Yes, OneStepLoopStep should certainly implement in interface as it is a service with public methods.  As for LoopStepInterface vs ILoopStep, this raises more questions and tech debt as there is no clearly defined style guideline.  The convention appears to suggest "LoopStepStrategy" or simply "LoopStep" -the former having my favour as it sounds more engaging from an enterprise perspective.

The biggest concern here is the lack of governance and review process that allowed this error to occur in the first place.  Who could even conceive of a world with a single loop impl?  I would not want to live in that world.

**Comment by vendethiel** [comment]
Maybe this should instead have a LoopStepFactory, with the int given as a parameter? or via DI
